### PR TITLE
Adds cypress and vitest

### PR DESCRIPTION
**Pull request by ryanlabouve**
Unit tests via `npm run dev`. Integration tests would work in a similar fashion:
CleanShot 2022-04-20 at 13 38 02

Acceptance tests via `npm run test:e2e:dev` in dev:
CleanShot 2022-04-20 at 13 38 50

### CODE REVIEW

File: app/config/networks/mainnet.test.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import mainnet from "./mainnet";
+test("mainnet has contracts", () => {
```
Comment (ryanlabouve) [inline app/config/networks/mainnet.test.ts:2]: Just a sample test

File: cypress/support/commands.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+       *    cy.cleanupUser()
+       * @example
+       *    cy.cleanupUser({ email: 'whatever@example.com' })
+       */
+      cleanupUser: typeof cleanupUser;
+    }
+  }
+}
+
+function login({
```
Comment (ryanlabouve) [inline cypress/support/commands.ts:33]: Need to port this for metmask connecting wallet.

File: cypress/support/create-user.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+// as that new user.
+
+import { parse } from "cookie";
+import { installGlobals } from "@remix-run/node/globals";
+import { createUserSession } from "~/session.server";
+import { createUser } from "~/models/user.server";
+
+installGlobals();
+
+async function createAndLogin(email: string) {
```
Comment (ryanlabouve) [inline cypress/support/create-user.ts:14]: Need to port this

File: cypress/support/commands.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import faker from "@faker-js/faker";
```
Comment (ryanlabouve) [inline cypress/support/commands.ts:1]: Leaving this as an example of how to add command

File: package.json [hunk 5]
```diff
@@ -11,24 +11,42 @@
     "dev": "run-p dev:*",
     "dev:css": "npm run generate:css -- --watch",
     "dev:remix": "cross-env NODE_ENV=development remix dev",
+    "format": "prettier --write .",
```
Comment (ryanlabouve) [inline package.json:14]: Adding all these to CI in a follow up

**Review by johnmosesman — APPROVED** [review]
where should tests live? under a test/ folder?

**Comment by ryanlabouve** [comment]
> where should tests live? under a test/ folder?

I prefer to have them colocated with the files they are testing. E.g. `/a/b/c/dongle.ts => /a/b/c/dongle.test.ts`. Happy to adjust in the future.

**Pull request merged**